pkg/azure/utils.go

```diff
@@ -157,21 +157,21 @@ func (d *MachinePlugin) generateDataDisks(vmName string, azureDataDisks []api.Az
 	return dataDisks
 }
 
-func (d *MachinePlugin) getVMParameters(vmName string, image *compute.VirtualMachineImage, networkInterfaceReferenceID string) compute.VirtualMachine {
+func (d *MachinePlugin) getVMParameters(vmName string, image *compute.VirtualMachineImage, networkInterfaceReferenceID string, providerSpec *api.AzureProviderSpec, secret *corev1.Secret) compute.VirtualMachine {
```

Comment:
Does this func now need the MachinePlugin?

```suggestion
func getVMParameters(vmName string, image *compute.VirtualMachineImage, networkInterfaceReferenceID string, providerSpec *api.AzureProviderSpec, secret *corev1.Secret) compute.VirtualMachine {
```